### ISSUE TITLE
Blin 2020.06, round 4

### ISSUE THREAD

**Issue by Altai-man** [rakudo/rakudo#3759]
[Blin](https://github.com/Raku/Blin) results between 2020.05.1 ([002acb1](https://github.com/rakudo/rakudo/commit/002acb1be2ba2a47ef8a48c30c340d43df91abed)) and HEAD ([1402c1d](https://github.com/rakudo/rakudo/commit/1402c1d1ed6b16eb04cc310dd99bc23d44e19df5)):

* [x] [IO::Glob](https://modules.raku.org/dist/IO::Glob) – Fail, Bisected: [b63976a](https://github.com/rakudo/rakudo/commit/b63976a8f19e650fe3533562a5fa040d84535de8)
  <details><Summary>Old Output</summary>

  ```
  ===> Searching for: IO::Glob
  ===> Found: IO::Glob:ver<0.9.0>:auth<github:zostay> [via Zef::Repository::LocalCache]
  ===> Testing: IO::Glob:ver<0.9.0>:auth<github:zostay>
  [IO::Glob] t/absolute.t ..... ok
  [IO::Glob] t/dir.t .......... ok
  [IO::Glob] t/double-star.t .. ok
  [IO::Glob] t/iterator.t ..... ok
  [IO::Glob] t/meta.t ......... ok
  [IO::Glob] t/path-dir.t ..... ok
  [IO::Glob] t/smart-match.t .. ok
  [IO::Glob] t/sqlish.t ....... ok
  [IO::Glob] All tests successful.
  [IO::Glob] Files=8, Tests=91, 24 wallclock secs ( 0.12 usr  0.01 sys + 16.38 cusr  1.24 csys = 17.75 CPU)
  [IO::Glob] Result: PASS
  ===> Testing [OK] for IO::Glob:ver<0.9.0>:auth<github:zostay>
  ===> Installing: IO::Glob:ver<0.9.0>:auth<github:zostay>
  ===> Install [OK] for IO::Glob:ver<0.9.0>:auth<github:zostay>

  ```
  </details>
  <details>
  <summary>New Output</summary>

  ```
  ===> Searching for: IO::Glob
  ===> Found: IO::Glob:ver<0.9.0>:auth<github:zostay> [via Zef::Repository::LocalCache]
  ===> Testing: IO::Glob:ver<0.9.0>:auth<github:zostay>
  [IO::Glob]     # Failed test at t/absolute.t line 21
  [IO::Glob]     # expected: $["Blin", "bin", "boot", "dev", "entrypoint.sh", "etc", "home", "lib", "lib64", "media", "mnt", "opt", "pkg-dependencies", "proc", "root", "run", "sbin", "srv", "sys", "tmp", "usr", "var"]
  [IO::Glob]     #      got: $[".", "..", "Blin", "bin", "boot", "dev", "entrypoint.sh", "etc", "home", "lib", "lib64", "media", "mnt", "opt", "pkg-dependencies", "proc", "root", "run", "sbin", "srv", "sys", "tmp", "usr", "var"]
  [IO::Glob]     # You failed 1 test of 1
  [IO::Glob] # Failed test 'root-in-the-glob'
  [IO::Glob] # at t/absolute.t line 19
  [IO::Glob]     # Failed test at t/absolute.t line 26
  [IO::Glob]     # expected: $["Blin", "bin", "boot", "dev", "entrypoint.sh", "etc", "home", "lib", "lib64", "media", "mnt", "opt", "pkg-dependencies", "proc", "root", "run", "sbin", "srv", "sys", "tmp", "usr", "var"]
  [IO::Glob]     #      got: $[".", "..", "Blin", "bin", "boot", "dev", "entrypoint.sh", "etc", "home", "lib", "lib64", "media", "mnt", "opt", "pkg-dependencies", "proc", "root", "run", "sbin", "srv", "sys", "tmp", "usr", "var"]
  [IO::Glob]     # You failed 1 test of 1
  [IO::Glob] # Failed test 'root-in-the-dir'
  [IO::Glob] # at t/absolute.t line 24
  [IO::Glob]     # Failed test at t/absolute.t line 31
  [IO::Glob]     # expected: $["Blin", "bin", "boot", "dev", "entrypoint.sh", "etc", "home", "lib", "lib64", "media", "mnt", "opt", "pkg-dependencies", "proc", "root", "run", "sbin", "srv", "sys", "tmp", "usr", "var"]
  [IO::Glob]     #      got: $[]
  [IO::Glob]     # You failed 1 test of 1
  [IO::Glob] # Failed test 'root-via-accept'
  [IO::Glob] # at t/absolute.t line 29
  [IO::Glob] # You failed 3 tests of 4
  [IO::Glob] t/absolute.t ..... 
  [IO::Glob] Dubious, test returned 3 (wstat 768, 0x300)
  [IO::Glob] Failed 3/4 subtests 
  [IO::Glob] t/dir.t .......... ok
  [IO::Glob] t/double-star.t .. ok
  [IO::Glob] t/iterator.t ..... ok
  [IO::Glob] t/meta.t ......... ok
  [IO::Glob] t/path-dir.t ..... ok
  [IO::Glob] t/smart-match.t .. ok
  [IO::Glob] t/sqlish.t ....... ok
  [IO::Glob] Test Summary Report
  [IO::Glob] -------------------
  [IO::Glob] t/absolute.t   (Wstat: 768 Tests: 4 Failed: 3)
  [IO::Glob]   Failed tests:  2-4
  [IO::Glob]   Non-zero exit status: 3
  [IO::Glob] Files=8, Tests=91, 29 wallclock secs ( 0.11 usr  0.03 sys + 22.61 cusr  1.51 csys = 24.26 CPU)
  [IO::Glob] Result: FAIL
  ===> Testing [FAIL]: IO::Glob:ver<0.9.0>:auth<github:zostay>
  [IO::Glob] Failed to get passing tests, but continuing with --force-test
  ===> Installing: IO::Glob:ver<0.9.0>:auth<github:zostay>
  ===> Install [OK] for IO::Glob:ver<0.9.0>:auth<github:zostay>

  ```
  </details>

| Status                    | Count |          Modules          |
| :------------------------ | :---: | :------------------------ |
| UnhandledException        |     1 | [Foo::�Foo](https://modules.raku.org/dist/Foo::�Foo) |
| Flapper                   |     3 | [Cro::WebSocket](https://modules.raku.org/dist/Cro::WebSocket) [JSON::simd](https://modules.raku.org/dist/JSON::simd) [Test::Async](https://modules.raku.org/dist/Test::Async) |
| Fail                      |     5 | [CoreHackers::Sourcery](https://modules.raku.org/dist/CoreHackers::Sourcery) [IO::Glob](https://modules.raku.org/dist/IO::Glob) |
| ZefFailure                |     7 | [Concurrent::BoundedChannel](https://modules.raku.org/dist/Concurrent::BoundedChannel) [Cro::ZeroMQ](https://modules.raku.org/dist/Cro::ZeroMQ) [DateTime::TimeZone](https://modules.raku.org/dist/DateTime::TimeZone) [Games::BubbleBreaker](https://modules.raku.org/dist/Games::BubbleBreaker) [Gnome::Gtk3](https://modules.raku.org/dist/Gnome::Gtk3) [MongoDB](https://modules.raku.org/dist/MongoDB) [Pretty::Table](https://modules.raku.org/dist/Pretty::Table) |
| InstallableButUntested    |     7 | [HTTP::Server::Async](https://modules.raku.org/dist/HTTP::Server::Async) [IO::Socket::Async::SSL](https://modules.raku.org/dist/IO::Socket::Async::SSL) [IRC::Client](https://modules.raku.org/dist/IRC::Client) [Log::Minimal](https://modules.raku.org/dist/Log::Minimal) [Time::Duration](https://modules.raku.org/dist/Time::Duration) [Toaster](https://modules.raku.org/dist/Toaster) [Uzu](https://modules.raku.org/dist/Uzu) |
| MissingDependency         |     8 | [CSS::Module](https://modules.raku.org/dist/CSS::Module) [CSS::Selector::To::XPath](https://modules.raku.org/dist/CSS::Selector::To::XPath) [DBIx::NamedQueries](https://modules.raku.org/dist/DBIx::NamedQueries) [HTTP::Server::Middleware::JSON](https://modules.raku.org/dist/HTTP::Server::Middleware::JSON) [HTTP::Server::Router](https://modules.raku.org/dist/HTTP::Server::Router) [HTTP::Server::Router::YAML](https://modules.raku.org/dist/HTTP::Server::Router::YAML) [Task::Galaxy](https://modules.raku.org/dist/Task::Galaxy) [Termbox](https://modules.raku.org/dist/Termbox) |
| CyclicDependency          |    13 | [Algorithm::HierarchicalPAM](https://modules.raku.org/dist/Algorithm::HierarchicalPAM) [Algorithm::LDA](https://modules.raku.org/dist/Algorithm::LDA) [Algorithm::LibSVM](https://modules.raku.org/dist/Algorithm::LibSVM) [CSS](https://modules.raku.org/dist/CSS) [CSS::Properties](https://modules.raku.org/dist/CSS::Properties) [Chart::Gnuplot](https://modules.raku.org/dist/Chart::Gnuplot) [Foo::Dependencies::A-on-B](https://modules.raku.org/dist/Foo::Dependencies::A-on-B) [Foo::Dependencies::B-on-A](https://modules.raku.org/dist/Foo::Dependencies::B-on-A) [Foo::Dependencies::Self](https://modules.raku.org/dist/Foo::Dependencies::Self) [Geo::Hash](https://modules.raku.org/dist/Geo::Hash) [HTML::Canvas](https://modules.raku.org/dist/HTML::Canvas) [HTML::Canvas::To::PDF](https://modules.raku.org/dist/HTML::Canvas::To::PDF) [MeCab](https://modules.raku.org/dist/MeCab) |
| AlwaysFail                |   310 | ⋯                         |
| OK                        |  1123 | ⋯                         |

This run started on 2020-06-17T19:38:12Z and finished in ≈7 hours.

**Comment by lizmat** [comment]
Again, `CoreHackers::Sourcery` is fixed, Blin is testing the wrong version.

**Comment by Altai-man** [comment]
Oops, sorry. You are right. Then a single one is left.

**Comment by JJ** [comment]
IO::Glob is fixed [here](https://github.com/JJ/perl6-IO-Glob) and waiting for the PR to be accepted. I couldn't find a way to pass the tests other than modify the tests, although there was a combination of genuine problems and incompatibility of tests. I think I got both, but anyway that distribution is undertested and thus underspecified.  I've added also a few more tests, but many more are needed.

**Comment by JJ** [comment]
Also, HTTP::Server::Router was [PR'd here](https://github.com/tony-o/perl6-http-server-router/pulls). The other HTTP modules depend on it. I don't think that's critical, but you can mark them as done.

**Comment by Altai-man** [comment]
>I couldn't find a way to pass the tests other than modify the tests

There is still a regression in rakudo indicated by this module. If you modify a test in the module you know, code bases we don't have access to still have chances to be suddenly broken.

**Comment by JJ** [comment]
Fixed after 6b22c1eb4d by @lizmat :100:

**Comment by Altai-man** [comment]
Closing. I'll do another round on Friday's night. Thanks for all the efforts!
